### ISSUE TITLE
There is no way to kill focus for the particular cell during changing focus

### ISSUE THREAD

**Issue by narendirasha** [angular-ui/ng-grid#3750]
Hi,

I am working on auto tabbing functionality. If the user enters a particular number of values in the focused cell editor (say 4 for ex), during the 4th digit entry the focus should go to next cell horizontally.
I can able to set the focus using below code
$scope.gridApi.cellNav.scrollToFocus( $scope.gridOptions.data[rowIndex], $scope.gridOptions.columnDefs[colIndex]);

But the previous cell is still not losing the focus and still editable. I have tried various options to kill the previous cell focus, but nothing worked.

Options I tried and failed:
1. $scope.gridApi.grid.cellNav.clearFocus();
2. $scope.gridOptions.columnDefs.cellEditableCondition = false;
3. $scope.gridOptions.columnDefs.enableCellEdit = false;
4. $scope.$apply();
5. $scope.gridApi.grid.refresh();

I tried using below

$scope.gridApi.grid.cellNav.broadcastCellNav(uiGridEditConstants.events.END_CELL_EDIT);
but getting "uid" error. 

The bottom line is how can I trigger "END_CELL_EDIT" event and commit the user entry in the cell before changing the focus to next one ? or 
any other work around is there to achieve this?

Thanks,
Naren

**Comment by swalters** [comment]
try to stop the edit in your editor when on the last character and then scroll in a timeout
  $scope.$emit(uiGridEditConstants.events.END_CELL_EDIT);
  //Put the scrollToFocus in a timeout so it occurs on the next event loop
// make sure your column is the next column
  $timeout(function(){
       $scope.gridApi.cellNav.scrollToFocus( $scope.gridOptions.data[rowIndex],  $scope.gridOptions.columnDefs[colIndex]);
   });

**Comment by narendirasha** [comment]
Hi Swalter,

Thanks for your inputs. the "emit" not working for me but 'broadcast' worked well. Also I replaced my custom template content with the default "cellEditor" template and added additional events and it seems the focus out of cell working to be fine now. :)

 found that I was missing "ui-grid-editor" directive in my custom template before which was in default editor.

Thanks a lot once again.
Naren

I found that I was missing "ui-grid-editor" directive in my custom template before.
